great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -402,6 +484,53 @@ def add_splitter_divided_integer(self, column_name: str, divisor: int) -> _SQLAs
             )
         )
 
+    def add_splitter_mod_integer(self, column_name: str, mod: int) -> _SQLAsset:
+        return self._add_splitter(
+            ColumnSplitterModInteger(
+                method_name="split_on_mod_integer",
+                column_name=column_name,
+                mod=mod,
+            )
+        )
+
+    def add_splitter_hashed_column(
+        self, column_name: str, hash_digits: int
+    ) -> _SQLAsset:
+        from great_expectations.experimental.datasources.sqlite_datasource import (
+            SqliteDatasource,
+        )
+
+        if not isinstance(self._datasource, SqliteDatasource):
```

Comment:
This isn't ideal but is the cleanest way to fail early here. A similar exception is raised inside of the `sqlalchemy_data_splitter.py` module and I wanted to raise earlier. Alternatives:

* raise late
* build out some sqlite specific infrastructure. Since this is at the `_SQLAsset` level, it is more involved than if it was at the `Datasource` level. While doable, it was more of a refactor than I wanted to do at the moment.

Reply:
Maybe the refactor is worth it if we don't want this to appear in autocomplete for postgres and other sql backends?

Reply:
Yeah, I've converted this PR to a draft and will fix this now.